### PR TITLE
[ET-VK] Enable Vulkan tests in CI

### PR DESCRIPTION
**Pull request by SS-JIA**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #3737

## Context

As title; enables Vulkan delegate to build and test during the unit test workflow.

Differential Revision: [D57968974](https://our.internmc.facebook.com/intern/diff/D57968974)

### CODE REVIEW

**Review by kirklandsign — APPROVED** [review]

File: .ci/scripts/setup-vulkan-linux-deps.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
+
+#!/bin/bash
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+set -ex
```
Comment (kirklandsign) [inline .ci/scripts/setup-vulkan-linux-deps.sh:9]: BTW is this script used for developers (non CI) as well?

Comment (SS-JIA) [reply .ci/scripts/setup-vulkan-linux-deps.sh:9]: nope, just for CI

**Comment by SS-JIA** [comment]
@SS-JIA has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D57968974).

**Comment by SS-JIA** [comment]
@SS-JIA has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D57968974).

**Pull request closed without merging**